internal/services/servicebus/servicebus_namespace_customer_managed_key_resource.go

```diff
@@ -0,0 +1,205 @@
+func resourceServiceBusNamespaceCustomerManagedKeyRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ServiceBus.NamespacesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.NamespaceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	if resp.SBNamespaceProperties == nil || resp.SBNamespaceProperties.Encryption == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("namespace_id", id.ID())
+	d.Set("infrastructure_encryption_enabled", resp.SBNamespaceProperties.Encryption.RequireInfrastructureEncryption)
+	if keyVaultProperties := resp.SBNamespaceProperties.Encryption.KeyVaultProperties; keyVaultProperties != nil && len(*keyVaultProperties) != 0 {
+		props := (*keyVaultProperties)[0]
+		keyVaultKeyId, err := keyVaultParse.NewNestedItemID(*props.KeyVaultURI, "keys", *props.KeyName, *props.KeyVersion)
+		if err != nil {
+			return fmt.Errorf("parsing `key_vault_key_id`: %+v", err)
+		}
+		d.Set("key_vault_key_id", keyVaultKeyId.ID())
+		if props.Identity != nil {
+			d.Set("identity_id", props.Identity.UserAssignedIdentity)
+		}
+	}
+
+	return nil
+}
+
+func resourceServiceBusNamespaceCustomerManagedKeyDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ServiceBus.NamespacesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.NamespaceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	locks.ByName(id.Name, "azurerm_servicebus_namespace")
+	defer locks.UnlockByName(id.Name, "azurerm_servicebus_namespace")
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	// Since this isn't a real object and it cannot be disabled once Customer Managed Key at rest has been enabled
+	// And it must keep at least one key once Customer Managed Key is enabled
+	// So for the delete operation, it has to recreate the EventHub Namespace with disabled Customer Managed Key
+	deleteFuture, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting %s: %+v", *id, err)
+	}
+	if err = deleteFuture.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(deleteFuture.Response()) {
+			return fmt.Errorf("failed to wait for removal of %q: %+v", id, err)
+		}
+	}
+
+	namespace := resp
+	namespace.Encryption = nil
+
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, namespace)
+	if err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for create/update of %s: %+v", id, err)
+	}
```

Comment:
Unfortunately this isn't something we can ship - this deletes all messages and other configuration for the ServiceBus Namespace with no confirmation.

Can you reach out to the Service Team about this?

Reply:
Unfortunately this feature can't be disabled, 

![image](https://user-images.githubusercontent.com/79895375/158094191-1c1fb5d8-083a-468c-98bd-ed4d61475a40.png)

Reply:
As suggested above, I'd recommend reaching out to the Service Team here.

Whilst we unintentionally shipped support for this in EventHub Namespace - in 3.0 this resources delete becomes a noop - which isn't ideal and is the only possible alternative instead of removing this resource entirely.

In order to ship this, we'll need a means of disabling this - so I'd suggest chatting with the Service Team here, as deleting all user data and configuration silently within the ServiceBus Namespace isn't something we can ship unfortunately.

Reply:
Hi @tombuildsstuff ,

I've confirmed with service team, that CMK can't be disabled and won't support disabling CMK in the future. I've added a a commit to make delete no-op. Please take another look, thanks!

Reply:
@ms-henglu since this can't be disabled we'll need to instead inline this within the ServiceBus Namespace resource - whilst that's not ideal (as it means this can't work for System Assigned Identities), it's the only way to make this work given the service limitations here